components/SubmissionComments.tsx

```diff
@@ -34,6 +39,46 @@ export const SubmissionComments: React.FC<{
     })
   }
 
+  const handleStartEditing = (index: number, initialInput: string) => {
+    if (editing !== undefined)
+      alert(
+        'You can only edit one comment in a single comment chain at a time.'
+      )
+    else {
+      setEditing(index)
+      setCurInput(initialInput)
+    }
+  }
+
+  const handleUpdateComment = async (c: Comment, newContent: string) => {
+    const update = updateCache({
+      submissionId: submission.id,
+      content: newContent,
+      lessonId: submission.lessonId,
+      challengeId: submission.challengeId,
+      userId: submission.user.id
+    })
+
+    await editComment({
+      variables: {
+        id: c.id,
+        content: newContent
+      },
+      update
+    })
+
+    setEditing(undefined)
+    setCurInput('')
+  }
+
+  const handleDiscardChange = () => {
+    setEditing(undefined)
+    setCurInput('')
+  }
+
+  //to make sure comments load in correct order
+  comments = _.orderBy(comments, ['id'], ['asc'])
```

Comment:
I'm not a fan of modifying the parameters. Perhaps I'm biased. But isn't there away to not modify the parameters? Like storing a local copy of the comments and modify it.

Reply:
perhaps a deepcopy of the comments?

Reply:
I don't think the `comments` is modified. Just a `const submissionsComments = comments` will do the job.